docs/guide/plugin-development.md

```diff
@@ -51,7 +51,13 @@ export default function myPlugin() {
 
 Rolldown can now compile and execute the regular expression on the Rust side, and can avoid invoking JS if the filter does not match.
 
-In addition to `id`, you can also filter based on `moduleType` and the module's source code. Full `HookFilter` interface for the `filter` property:
+In addition to `id`, you can also filter based on `moduleType` and the module's source code. The `filter` property works similarly to [`createFilter` from `@rollup/pluginutils`](https://github.com/rollup/plugins/blob/master/packages/pluginutils/README.md#createfilter). Here are some important details:
+
+- If multiple values are passed to `include`, the filter matches if **any** of them match.
+- If a filter has both `include` and `exclude`, `exclude` takes precedence.
+- The filters are checked in the following order: `moduleType`, `id`, `code`. If any filter matches, it **early returns**. For example, if a module is excluded by the `moduleType` filter, it is excluded regardless of the `id` or `code` filters.
```

Comment:
so
```
{
   moduleType: {include: 'js'},
   id: {exclude:/node_modules/} 
}
```

would not exclude js modules in node_modules?

i think it should try all includes and all excludes regardless and only pick it if include is true and exclude is false in the end. It can still do early breaks if it found the first match for either.

Reply:
yes. but yeah this feels hard to use...

Reply:
if filters are separate and order is relevant, i'd argue that it should be an array rather than a map to give users more control

Reply:
Nice suggestion, we will discussed it later.